packages/yourloops/pages/hcp/team-edit-modal.tsx

```diff
@@ -0,0 +1,240 @@
+    valid = valid && addrCountry.length > 1;
+    valid = valid && (teamEmail.length < 1 || REGEX_EMAIL.test(teamEmail));
+    return !valid;
+  };
+
+  const handleCloseModal = (): void => {
+    setModalOpen(false);
+  };
+  const handleValidateModal = (): void => {
+    setModalOpen(false);
+
+    team.name = teamName;
+    team.phone = teamPhone;
+
+    if (teamEmail.length > 0) {
+      team.email = teamEmail;
+    } else {
+      delete team.email;
+    }
+
+    team.address = {
+      line1: addrLine1,
+      line2: addrLine2,
+      zip: addrZipCode,
+      city: addrCity,
+      country: addrCountry,
+    };
+    if (addrLine2.length < 1) {
+      delete team.address?.line2;
+    }
+    onSaveTeam(team);
+  };
+  const handleChangeTeamName = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setTeamName(e.target.value);
+  };
+  const handleChangeTeamPhone = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setTeamPhone(e.target.value);
+  };
+  const handleChangeTeamEmail = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setTeamEmail(e.target.value);
+  };
+  const handleChangeAddrLine1 = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setAddrLine1(e.target.value);
+  };
+  const handleChangeAddrLine2 = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setAddrLine2(e.target.value);
+  };
+  const handleChangeAddrZip = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setAddrZipCode(e.target.value);
+  };
+  const handleChangeAddrCity = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    setAddrCity(e.target.value);
+  };
+  const handleChangeAddrCountry = (e: React.ChangeEvent<{ name?: string | undefined; value: unknown; }>): void => {
+    const country = e.target.value as string;
+    setAddrCountry(country);
+  };
+
+  if (formIsIncomplete !== isFormIsIncomplete()) {
+    setFormIsIncomplete(!formIsIncomplete);
+  }
+
+  return (
+    <Modal id="team-modal-edit" aria-labelledby={ariaModal} className={classes.modal} open={modalOpened} BackdropComponent={Backdrop} BackdropProps={{ timeout: modalBackdropTimeout }}>
+      <Fade in={modalOpened}>
+        <div className={classes.divMain}>
+          <h2 id="team-modal-edit-title">{modalTitle}</h2>
+          <form noValidate autoComplete="off" className={classes.form}>
+            <TextField id="team-modal-edit-field-name" className={classes.formChild} variant="outlined" onChange={handleChangeTeamName} name="name" value={teamName} label={t("team-modal-edit-placeholder-name")} required={true} aria-required="true" />
+
+            <TextField id="team-modal-edit-field-line1" className={classes.formChild} variant="outlined" onChange={handleChangeAddrLine1} name="addr-line1" value={addrLine1} label={t("team-modal-edit-placeholder-addr-line1")} required={true} aria-required="true" />
```

Comment:
Would that make sense to make it more readable that way ?

```suggestion
            <TextField 
              id="team-modal-edit-field-line1" 
              className={classes.formChild} 
              variant="outlined" 
              onChange={handleChangeAddrLine1} 
              name="addr-line1" 
              value={addrLine1} 
              label={t("team-modal-edit-placeholder-addr-line1")} 
              required={true} 
              aria-required="true" />
```
Same for the other similar fields.
Just for readibility ;-)

Reply:
Why not.